pkg/apis/operator/base/common.go

```diff
@@ -323,6 +331,14 @@ type EnvRequirementsOverride struct {
 	EnvVars []corev1.EnvVar `json:"envVars,omitempty"`
 }
 
+// ProbesRequirementsOverride enables the user to override any container's env vars.
+type ProbesRequirementsOverride struct {
+	// The container name
+	Container string `json:"container"`
+	// The desired ProbeRequirements
+	Probe corev1.Probe `json:"probe,omitempty"`
```

Comment:
Should we prevent setting the [`ProbeHandler` field](https://github.com/skonto/operator/blob/23892c03b5a5caeb2e41691a4bff41248903dd73/vendor/k8s.io/api/core/v1/types.go#L2462) and allow only these fields https://github.com/skonto/operator/blob/23892c03b5a5caeb2e41691a4bff41248903dd73/vendor/k8s.io/api/core/v1/types.go#L2194-L2226.

Reply:
I didnt want to limit people on how they can use this but we can.

Reply:
Let's not assume the options for users. If we plan to add some, what is the use case? Otherwise, limit to what is necessary. Drop ProbeHandler at least.

Reply:
@skonto